hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -109,8 +109,18 @@ extends Importer:
   def mkLetBinding(sym: LocalSymbol, rhs: Term): Ls[Statement] =
     LetDecl(sym) :: DefineVar(sym, rhs) :: Nil
   
-  def resolveField(srcTree: Tree, base: Opt[Symbol], nme: Ident): Opt[Symbol] =
+  def resolveField(srcTree: Tree, base: Opt[Symbol], nme: Ident): Ctxl[Opt[Symbol]] =
     base match
+    // Look up symbols qualified by `globalThis.`.
+    case S(tsym: TopLevelSymbol) =>
```

Comment:
Can you explain why this is needed?

Reply:
It fixes the issue where we can't get the class symbol when accessing a class in `Predef` through `globalThis`. I need to match terms against `MatchResult`. But I found:
```scala
MatchResult(()) is globalThis.Predef.MatchResult(_)
//│ FAILURE: Unexpected type error
//│ ╔══[ERROR] Cannot use this selection as an extractor
//│ ║  l.19: 	MatchResult(()) is globalThis.Predef.MatchResult(_)
//│ ╙──      	                   ^^^^^^^^^^
//│ = false
```
I followed the `cls` function and found the `resolveField` function, but it doesn't handle the case where base is `globalThis` (`TopLevelSymbol`).

But I just tried it again, and I realized the silly part of this code, the `find` function is redundant, and I will fix it.

Reply:
But what's the justification for writing such code as `is globalThis.Predef.MatchResult(_)` rather than `is Predef.MatchResult(_)`? And if it's not written by the user, you should be able to just add the correct symbol in the section term when constructing it...

Reply:
> But what's the justification for writing such code as `is globalThis.Predef.MatchResult(_)` rather than `is Predef.MatchResult(_)`?

To prevent `Predef` from being shadowed.

> you should be able to just add the correct symbol in the section term when constructing it...

Maybe this should be the right way.

Reply:
What on Earth are you talking about?

Reply:
Users are NEVER supposed/expected to make use of `globalThis`.

Reply:
Sorry for the confusion. The code in my initial response is the term I want to generate (I really don't need/expect the user to write it and I also never do that).

In this PR, I need to generate a term similar to `unapply(scrut) is MatchResult(…)` and `MatchResult` is defined in `Predef`. I know I can use `Predef.MatchResult`. But I was told that when generating terms, if I need to use things defined in `Predef`, I should prefix it with `globalThis` to prevent shadowing.

For example, if I need to use `tupleSlice`, the term should be `globalThis.Predef.tupleSlice`. So, to use `MatchResult`, I generate the term `globalThis.Predef.MatchResult`. But the problem is that this term doesn't have a class symbol, therefore it can't be used in pattern matching.

Reply:
Plenty of people might shadow the word "Predef". You can't just say this kind of things isn't important. Correctness is important.

I finally had to check your code myself to find that one place you're talking about because you won't just give me a link or line number. It appears to be this:

```scala
  protected lazy val matchResultClass =
    Sel(Sel(Ident("globalThis"), Ident("Predef")), Ident("MatchResult"))
```

This does not look right. You're not **_generating a term_** here, as you keep writing, you're **_generating a tree_** and then using the elaborator to transform that into a term for you.

If you were truly generating a term, as you're supposed to (trees are what users write, they're generally not meant to be generated, in normal circumstances), you'd be using `Term.Sel` (or rather `Term.SynthSel`), which _allows you to pass the relevant symbol_ at construction. That's why your roundabout approach is not needed and was confusing me.

Reply:
Thank you for the careful review and your detailed explanation. I'm sorry for not providing you with a link or line number. I'd like to ask, how can I obtain the symbols of `Predef` and the class I need?

For example, now I know I can construct the term in the following way. Where can I get `A` and `B`?

```scala
Term.Sel(
  prefix = Term.Sel(
    prefix = state.globalThisSymbol.ref(),
    nme = Ident("Predef")
  )(S(/* A */)), // <- needs the symbol for `Predef`
  nme = Ident("MatchResult")
)(S(/* B */)) // <- needs the symbol for class `MatchResult`
```

Of course, I know these symbols are all in `Ctx`, but how do I know that what I found is the exact `Predef` imported by the compiler and not the one shadowed by the user?

Reply:
That's a good question, and now I understand your problem better. Thanks for explaining it.

What's for sure is that you don't want to write something like `ctx.get(nme.name).flatMap(_.symbol)` in `resolveField`, as that's just incorrect.

What I would probably do is declare `MatchResult` in `Prelude.mls`, which is used to declare builtins, and then I would use add a line in `Ctx#Builtins`, which can get the corresponding builtin symbols.

However, this will not automatically be the same as the symbol that's eventually going to be created in the `Predef` module, so users won't be able to use `MatchResult` explicitly yet. We can fix this later.